### PR TITLE
ArchiveArticle refactor and tests

### PR DESCRIPTION
**Pull request by gnott**
Reference issue https://github.com/elifesciences/elife-bot/issues/827, `ArchiveArticle` activity now using `storage_context` instead of direct connections to S3. This also makes it easier to test being able to avoid mocking all the boto objects.

I could possibly create some failing tests too. I was happy to get to this point.

### CODE REVIEW

File: activity/activity_ArchiveArticle.py [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
 import json
-import boto
-import random
 import zipfile
 from datetime import datetime
-import os
-from boto.s3.key import Key
-from boto.s3.connection import S3Connection
-import settings as settings_lib
+from provider.storage_provider import storage_context
```
Comment (giorgiosironi) [inline activity/activity_ArchiveArticle.py]: good abstraction

File: tests/activity/test_activity_archive_article.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    return [file_name for file_name in os.listdir(folder) if not file_name.startswith('.')]
+
+
+class TestArchiveArticle(unittest.TestCase):
+    def setUp(self):
+        fake_logger = FakeLogger()
+        self.activity = activity_object(settings_mock, fake_logger, None, None, None)
+
+    def tearDown(self):
+        helpers.delete_files_in_folder(activity_test_data.ExpandArticle_files_dest_folder)
```
Comment (giorgiosironi) [inline tests/activity/test_activity_archive_article.py]: setup and teardown seem reasonably reused here :+1:

File: tests/activity/test_activity_archive_article.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        expected_zip_file_name = 'elife-00353-vor-v1-20121213000000.zip'
+        expected_zip_files = [
+            'elife-00353-fig1-v1.tif',
+            'elife-00353-v1.xml',
+            'elife-00353-v1.pdf']
+        test_destination_folder = activity_test_data.ExpandArticle_files_dest_folder
+        fake_storage_context.return_value = FakeStorageContext()
+        self.activity.emit_monitor_event = mock.MagicMock()
+
+        success = self.activity.do_activity(activity_test_data.data_example_before_publish)
```
Comment (giorgiosironi) [inline tests/activity/test_activity_archive_article.py]: the arrange, act, assert phases of the test are clearly distinguishable here

File: tests/activity/test_activity_archive_article.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        self.assertEqual(success, self.activity.ACTIVITY_SUCCESS)
+        self.assertEqual(len(outbox_files(test_destination_folder)), expected_outbox_count)
+        # should be one zip file
+        zip_file_path = os.path.join(
```
Comment (giorgiosironi) [inline tests/activity/test_activity_archive_article.py]: to keep the test readable a `_assertZip()` function could be extracted with the rest of this test's code

Comment (gnott) [reply tests/activity/test_activity_archive_article.py]: I can see the benefit, I'm structuring it slightly differently though. I should be able to add a failure test or two using it as well.

**Pull request merged**